src/middlewared/middlewared/plugins/gluster_linux/peer.py

```diff
@@ -1,26 +1,32 @@
-from glustercli.cli import peer
-from middlewared.async_validators import resolve_hostname
-from middlewared.schema import Dict, Str
-from middlewared.service import (accepts, private, job,
-                                 CallError, Service,
-                                 ValidationErrors)
-from .utils import GlusterConfig, run_method
-
-import subprocess
 import xml.etree.ElementTree as ET
+from glustercli.cli import peer
+
+from middlewared.utils import filter_list, run
+from middlewared.schema import Dict, Str, Bool
+from middlewared.service import (accepts, private, job, filterable,
+                                 CallError, CRUDService)
+from .utils import GlusterConfig
 
 
 GLUSTER_JOB_LOCK = GlusterConfig.CLI_LOCK.value
 
 
-class GlusterPeerService(Service):
+class GlusterPeerService(CRUDService):
 
     class Config:
         namespace = 'gluster.peer'
 
-    @private
-    def _parse_peer(self, p):
+    @filterable
+    async def query(self, filters=None, options=None):
```

Comment:
From your CLI changes, we should not use =None here, right @themylogin ?

Reply:
Yes, let's get rid of this, thank you, @william-gr

Reply:
Do I need to wait until the CLI changes are merged? @themylogin @william-gr

Reply:
No, you can merge it now